README.md

```diff
@@ -1,116 +1,220 @@
-shell with your prefix:
+The  `--docker` switch  (or short  `-d`) is  used to  select build  environment.
+Environment  configurations   are  stored  in  `./build-env/target/`.   To  list
+available environments run:
+```bash
+./build-env/bin/build-all.sh -l env
 ```
-export PELION_DOCKER_PREFIX=${USER}-
+
+The `-d`  accepts also  partial environments  names (`-d  rh`, `-d  rhel/8`, `-d
+rhel-8`,  `-d  rhel`  -  all  are  valid).  When  name  matches  more  than  one
+environment, script will print error:
+```bash
+$ ./build-env/bin/build-all.sh -d 8
+Unable to load environment: ambiguous environment name, matches:
+centos/8 rhel/8
+
 ```
 
-The system is configured to use `sudo` without a password.
+## Preparing host to compile packages
+### System requirements
+The `build-all.sh` Script requires docker, bash 4.2+ and gnu-getopt.
 
+To  build arm64  packages  for *Red  Hat*  or *Centos*  `qemu`  is required  and
+qemu-docker  integration  because  *Red  Hat*/*Centos* does  not  support  cross
+compiling.
 
-2. Run docker image:
-```
-$ ./build-env/bin/docker-run.sh <image name>
+To enable qemu-docker integration on Linux `binfmt` has to be enabled using this
+image before using arm64 compilation:
+```bash
+docker run --rm --privileged docker/binfmt:a7996909642ee92942dcd6cff44b9b95f08dad64
 ```
 
-For example:
-```
-$ ./build-env/bin/docker-run.sh pelion-bionic-source
+The `binfmt` has to be executed once after every reboot.
+
+Build on MacOS is not tested.
+
+### Red Hat Enterprise Linux (RHEL) subscription
+RHEL  images  does  not  include  subscription  credentials.  `RH_USERNAME`  and
+`RH_PASSWORD` variables  with your  Red Hat  subscription credentials  should be
+exported as environment variables before running build scripts:
+
+```bash
+export RH_USERNAME="Your Red Hat username"
+export RH_PASSWORD="Your Red Hat password"
 ```
 
-The root of this repo is mounted to `/pelion-build`.
+Variables will be used to create docker image. If any of the variables would not
+be set, scripts  will interactively ask for them. Credentials  will be stored in
+docker image. Note that for free development subscription only 16 systems can be
+registered (registered  systems can be  removed on RHEL  subscription management
+page).
 
-Please note that `docker-run.sh` script does not use `PELION_DOCKER_PREFIX`.
+To create Red Hat account visit: https://www.redhat.com/wapps/ugc/register.html
 
-### Build environment for debian-10
 
-1. Scripts for creating clean build docker images:
-```
-$ ./build-env/bin/docker-debian-buster-create.sh # Debian 10
+## Build environment
+
+To access  build environment console `docker-run-env.sh`  script was introduced.
+The script accepts options until first positional argument (which is environment
+name).
+
+Example usage:
+```bash
+# run new rhel/8 image for host arch:
+./build-env/bin/docker-run-env.sh rhel
+
+# run centos container (attach to existing)
```

Comment:
Yes, it will attach to running container, if it is not running it will be started. With `-c` you can attach to same container multiple times (useful for debugging/development)

Reply:
Could we use `exec` here instead of `attach`?  That seems to better match existing [docker terminology](https://docs.docker.com/engine/reference/commandline/exec/).

Reply:
Sure, updated.